inference-engine/src/inference_engine/cpp/ie_infer_request.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "cpp/ie_infer_request.hpp"
+#include "cpp_interfaces/interface/ie_iinfer_request_internal.hpp"
+#include "cpp_interfaces/base/ie_infer_async_request_base.hpp"
+#include "ie_remote_context.hpp"
+
+namespace InferenceEngine {
+
+#define CATCH_IE_EXCEPTION(ExceptionType) catch (const InferenceEngine::ExceptionType& e) {throw e;}
+
+#define CATCH_IE_EXCEPTIONS                     \
+        CATCH_IE_EXCEPTION(GeneralError)        \
+        CATCH_IE_EXCEPTION(NotImplemented)      \
+        CATCH_IE_EXCEPTION(NetworkNotLoaded)    \
+        CATCH_IE_EXCEPTION(ParameterMismatch)   \
+        CATCH_IE_EXCEPTION(NotFound)            \
+        CATCH_IE_EXCEPTION(OutOfBounds)         \
+        CATCH_IE_EXCEPTION(Unexpected)          \
+        CATCH_IE_EXCEPTION(RequestBusy)         \
+        CATCH_IE_EXCEPTION(ResultNotReady)      \
+        CATCH_IE_EXCEPTION(NotAllocated)        \
+        CATCH_IE_EXCEPTION(InferNotStarted)     \
+        CATCH_IE_EXCEPTION(NetworkNotRead)      \
+        CATCH_IE_EXCEPTION(InferCancelled)
+
+#define CALL_STATEMENT(...)                                                                        \
```

Comment:
I'm getting CI failure now, probably due to this recent change:

In file included from inference-engine/src/inference_engine/CMakeFiles/inference_engine_obj.dir/Unity/unity_0_cxx.cxx:9:0:
/agent/_work/1/openvino/inference-engine/src/inference_engine/cpp/ie_infer_request.cpp:33:0: error: "CALL_STATEMENT" redefined [-Werror]
 #define CALL_STATEMENT(...)                                                                        \
 
In file included from inference-engine/src/inference_engine/CMakeFiles/inference_engine_obj.dir/Unity/unity_0_cxx.cxx:8:0:
/agent/_work/1/openvino/inference-engine/src/inference_engine/cpp/ie_executable_network.cpp:14:0: note: this is the location of the previous definition
 #define CALL_STATEMENT(...)                                                                        \
 
cc1plus: all warnings being treated as errors

Reply:
Fixed here https://github.com/openvinotoolkit/openvino/pull/5309#event-4612858830